Projects/DbCLI/SearchDb.py

```diff
@@ -509,116 +511,122 @@ def poolFromCurs(curs, similarityMethod):
+                    help='export a smiles file with the matching molecules')
+  parser.add_argument('--nonchiralSmiles', dest='chiralSmiles', default=True, action='store_false',
+                    help='do not use chiral SMILES in the output')
+  parser.add_argument('--silent', default=False, action='store_true',
+                    help='Do not generate status messages.')
+  
+  parser.add_argument('--zipMols', '--zip', default=False, action='store_true',
+                    help='read compressed mols from the database')
+  
+  parser.add_argument('--pharm2DTableName', default='pharm2dfps',
+                    help='name of the Pharm2D fingerprints table')
+  parser.add_argument('--fdefFile', '--fdef',
+                    default=os.path.join(RDConfig.RDDataDir, 'Novartis1.fdef'),
+                    help='provide the name of the fdef file to use for 2d pharmacophores')
+  parser.add_argument('--gobbi2DTableName', default='gobbi2dfps',
+                    help='name of the Gobbi2D fingerprints table')
+  
+  parser.add_argument(
+    '--similarityType', '--simType', '--sim', default='RDK', choices=supportedSimilarityMethods,
+    help='Choose the type of similarity to use, possible values: RDK, AtomPairs, TopologicalTorsions, Pharm2D, Gobbi2D, Avalon, Morgan. The default is %(default)s')
+  parser.add_argument('--morganFpDbName', default='Fingerprints.sqlt',
+                    help='name of the morgan fingerprints database')
+  parser.add_argument('--morganFpTableName', default='morganfps',
+                    help='name of the morgan fingerprints table')
+  parser.add_argument('--morganFpColName', default='morganfp',
+                    help='name of the morgan fingerprint column')
+  
+  parser.add_argument(
+    '--similarityMetric', '--simMetric', '--metric', default='',
+    choices=('tanimoto', 'dice', 'tversky', ''),
+    help='Choose the type of similarity to use, possible values: tanimoto, dice, tversky. The default is determined by the fingerprint type')
+  parser.add_argument('--tverskyA', default=0.5, type=float, help='Tversky A value')
+  parser.add_argument('--tverskyB', default=0.5, type=float, help='Tversky B value')
+  parser.add_argument(
+    '--simThresh', default=-1, type=float,
+    help='threshold to use for similarity searching. If provided, this supersedes the topN argument')
+  return parser
 
 if __name__ == '__main__':
-  import sys, getopt, time
 
-  options, args = parser.parse_args()
-  if len(args) != 1 and not (options.smilesQuery or options.smartsQuery or options.propQuery):
+  parser = initParser()
+  options = parser.parse_args()
+  
+  if options.filename is None and not (options.smilesQuery or options.smartsQuery or options.propQuery):
     parser.error('please either provide a query filename argument or do a data or smarts query')
 
-  if len(args):
-    queryFilename = args[0]
-  else:
-    queryFilename = None
+  queryFilename = options.filename
```

Comment:
No need to use the if-statement. If filename isn't given on the command line, `options.filename` is `None`